Shopify/cli
# Fixed Live reload no longer works for extensions

### WHY are these changes introduced?

Fixes https://github.com/Shopify/cli/issues/345
Fixes https://github.com/Shopify/ui-extensions/issues/449

TBD: 
- Problem with reloading the locale content automatically could not be reproduced. To be decided if another issue should be created for that problem

### WHAT is this pull request doing?
- Fixes the configuration of the esBuild watch handler. Now the go extensions binary receives the messages when the extension is rebuild

### How to test your changes?
1. Scaffold a Checkout UI extension
2. Run yarn dev
3. Click on the Checkout UI extension preview link
4. Make changes to the src/index.tsx
5. Observe that the local extension is reloaded
